packages/supi/src/api/storeAdd.ts

```diff
@@ -0,0 +1,53 @@
+import normalizeRegistryUrl = require('normalize-registry-url')
+
+import {streamParser} from '@pnpm/logger'
+import {StoreController} from 'package-store'
+
+import parseWantedDependencies from '../parseWantedDependencies';
+import {ReporterFunction} from '../types'
+
+export default async function (
+  fuzzyDeps: string[],
+  opts: {
+    registry?: string,
+    verifyStoreIntegrity?: boolean,
+    reporter?: ReporterFunction,
+    storeController: StoreController,
+    tag?: string,
+  },
+) {
+  const reporter = opts && opts.reporter
+  if (reporter) {
+    streamParser.on('data', reporter)
+  }
+
+  const deps = await parseWantedDependencies(fuzzyDeps, {
+    allowNew: true,
+    currentPrefs: {},
+    defaultTag: opts.tag || 'default',
+    dev: false,
+    devDependencies: {},
+    optional: false,
+    optionalDependencies: {},
+  })
+
+  await Promise.all(deps.map(async (dep) => {
+    const pkgResponse = await opts.storeController.requestPackage(dep, {
+      downloadPriority: 1,
+      loggedPkg: {
+        rawSpec: dep.raw,
+      },
+      preferredVersions: {},
+      prefix: '',
+      registry: normalizeRegistryUrl(opts.registry || 'https://registry.npmjs.org/'),
+      verifyStoreIntegrity: opts.verifyStoreIntegrity || true,
+    })
+    await pkgResponse['fetchingFiles'].catch((err: Error) => {}) // tslint:disable-line
```

Comment:
I think there should be an info message printed about the added tarballs.
When you run `pnpm store add typescript@1` it would be useful to see which version of typescript was added. So maybe just print the pkgId?

```
$ pnpm store add typescript@1
+ registry.npmjs.org/typescript/1.2.1
```

`pnpm store prune` provides a similar output at the moment.

Also, I think it is important to print failures. So don't just ignore if `fetchingFiles` fails. Maybe the process should exit with a non-zero exit code. Fetching the package is the intention of the operation after all, so if it fails, the operation was not successful.